test/test_dropbox.py

```diff
@@ -137,5 +144,57 @@ def test_team(self, dbxt):
             team_member_id = r.members[0].profile.team_member_id
             dbxt.as_user(team_member_id).files_list_folder('')
 
+    @dbx_from_env
+    def test_with_path_root_constructor(self, dbx):
+        # Verify valid mode types
+        for mode, ns_id in (
+                ("home", None),
+                ("root", "123"),
+                ("namespace_id", "123")
+        ):
+            dbx_new = dbx.with_path_root(mode, ns_id)
+            self.assertIsNot(dbx_new, dbx)
+
+            path_root_mode = getattr(PathRoot, mode)
+            path_root = path_root_mode(ns_id) if ns_id else path_root_mode
+
+            expected = stone_serializers.json_encode(PathRoot_validator, path_root)
+            self.assertEqual(dbx_new._headers.get(PATH_ROOT_HEADER), expected)
+
+        # verify invalid mode raises ValueError
+        with self.assertRaises(ValueError):
+            dbx.with_path_root("invalid_mode")
+
+    @dbx_from_env
+    def test_path_root(self, dbx):
+        root_info = dbx.users_get_current_account().root_info
+        root_ns = root_info.root_namespace_id
+        home_ns = root_info.home_namespace_id
+
+        # verify home mode
+        dbxpr = dbx.with_path_root('home')
+        dbxpr.files_list_folder('')
+
+        # verify root mode
+        dbxpr = dbx.with_path_root("root", root_ns)
+        dbxpr.files_list_folder('')
+
+        # verify namespace_id mode
+        dbxpr = dbx.with_path_root("namespace_id", home_ns)
+        dbxpr.files_list_folder('')
+
+    @dbx_from_env
+    def test_path_root_err(self, dbx):
+        # verify invalid namespace return is_no_permission error
+        dbxpr = dbx.with_path_root("namespace_id", "1234567890")
```

Comment:
I wonder if this NS will ever exist…. 🤔😏

Reply:
doesn't matter if it does or not, the error is consistent.  The only way this would fail is if the test account has access to that namespace.